### PR TITLE
Support Python 2.7

### PR DESCRIPTION
**Pull request by pquentin**
(And various CI fixes)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -8,11 +8,11 @@ environment:
 
   matrix:
     # XX TODO: re-enable
```
Comment (RatanShreshtha) [inline appveyor.yml]: No need of todo here as py 2.7 is enable

File: dev-requirements.txt [hunk 2]
```diff
@@ -5,20 +5,22 @@ wheel==0.30.0
...
 pytest-cov==2.6.0
 tox==2.9.1
 h11==0.8.0
 cryptography==2.2.1
-trio==0.3.0
+# https://github.com/GoogleCloudPlatform/python-repo-tools/issues/23
+pylint<2.0;python_version<="2.7"
+gcp-devrel-py-tools==0.0.15
+
+# optional dependencies, only intended for use with Python 3.5+
```
Comment (RatanShreshtha) [inline dev-requirements.txt]: Let's group optional dependencies, only intended for use with Python 3.5+ similarly for python 2

File: appveyor.yml (hunk 1 above)
Comment (pquentin) [reply appveyor.yml]: But Python 3.4 is still missing :) (Even though we want to remove it eventually.)

Comment (RatanShreshtha) [reply appveyor.yml]: lets just change the position of todo for py34

File: dev-requirements.txt (hunk 2 above)
Comment (pquentin) [reply dev-requirements.txt]: I don't understand, what are you suggesting? The Python 3.5+ dependencies are trio and twisted and are grouped below

File: appveyor.yml (hunk 1 above)
Comment (pquentin) [reply appveyor.yml]: Thanks, done

**Comment by pquentin** [comment]
AppVeyor is now failing and I don't know why 😞

**Comment by pquentin** [comment]
OK, the only difference was that the latest build picked up tox 3.10.0 instead of tox 3.9.0. tox 3.10.0 stops updating pip on installation, so we got an old pip that does not support pyproject.toml, so unasync did not exist when installing the project. I pinned tox to 3.9.0 for now.

**Pull request merged**